src/lang/std/sketch.ts

```diff
@@ -1138,51 +1141,67 @@ export const circle: SketchLineHelper = {
     if (input.type !== 'arc-segment') return ARC_SEGMENT_ERR
     const { center, radius } = input
     const _node = { ...node }
-    const nodeMeta = getNodeFromPath<CallExpression>(_node, pathToNode)
+    const nodeMeta = getNodeFromPath<CallExpressionKw>(_node, pathToNode)
     if (err(nodeMeta)) return nodeMeta
 
-    const { node: callExpression, shallowPath } = nodeMeta
+    const { node: callExpression } = nodeMeta
 
-    const firstArg = callExpression.arguments?.[0]
     const newCenter = createArrayExpression([
       createLiteral(roundOff(center[0])),
       createLiteral(roundOff(center[1])),
     ])
-    mutateObjExpProp(firstArg, newCenter, 'center')
+    mutateKwArg(ARG_CIRCLE_CENTER, callExpression, newCenter)
     const newRadius = createLiteral(roundOff(radius))
-    mutateObjExpProp(firstArg, newRadius, 'radius')
+    mutateKwArg(ARG_CIRCLE_RADIUS, callExpression, newRadius)
     return {
       modifiedAst: _node,
-      pathToNode: shallowPath,
+      pathToNode,
     }
   },
-  getTag: getTag(),
-  addTag: addTag(),
-  getConstraintInfo: (callExp: CallExpression, code, pathToNode) => {
-    if (callExp.type !== 'CallExpression') return []
+  getTag: getTagKwArg(),
+  addTag: addTagKw(),
+  getConstraintInfo: (callExp: CallExpressionKw, code, pathToNode) => {
+    if (callExp.type !== 'CallExpressionKw') return []
     const firstArg = callExp.arguments?.[0]
-    if (firstArg.type !== 'ObjectExpression') return []
-    const centerDetails = getObjExprProperty(firstArg, 'center')
-    const radiusDetails = getObjExprProperty(firstArg, 'radius')
-    if (!centerDetails || !radiusDetails) return []
-    if (centerDetails.expr.type !== 'ArrayExpression') return []
+    if (firstArg.type !== 'LabeledArg') return []
+    let centerDetails = null
+    let radiusDetails = null
+    let centerIndex = 0
+    let radiusIndex = 0
+    let index = 0
+    for (const prop of callExp.arguments) {
```

Comment:
I made these helper functions which should simplify this: https://github.com/KittyCAD/modeling-app/blob/main/src/lang/util.ts#L126-L164

Reply:
oh nice!